File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 9/16/2025, 12:08:57 AM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy=prices[0]
        maxProfit=0

        for x in range(1,len(prices)):
            if prices[x] > buy:
                profit=prices[x]-buy
                maxProfit=max(profit,maxProfit)
            else:
                buy=prices[x] 

        return maxProfit        

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 9/16/2025, 12:08:51 AM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit=0;
        int mini=prices[0];

        for(auto &it: prices){
            maxProfit=max(maxProfit,it-mini);
            mini=min(mini,it);
        }
        return maxProfit;
    }
};